src/shray.h

```diff
@@ -156,3 +156,13 @@ extern Heap heap;
             gasnet_exit(1);                                                   \
         }                                                                     \
     }
+
+static inline void MadviseDontNeedSafe(void *addr, size_t size) {
+  int r = madvise(addr, size, prot);
```

Comment:
`prot` is undeclared here, should probably be `MADV_DONTNEED`.

Reply:
Done.  (I don't have a good test setup to even make sure this stuff compiles.  Sorry.)